eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/EwonTagQuality.java

```diff
@@ -0,0 +1,278 @@
+      qualityInt = DMWEB_QUALITY_UNCERTAIN;
+    } else if (quality.equalsIgnoreCase(DMWEB_QUALITY_INITIAL_GOOD_STRING)) {
+      qualityInt = DMWEB_QUALITY_INITIAL_GOOD;
+    } else if (quality.equalsIgnoreCase(DMWEB_QUALITY_UNKNOWN_STRING)) {
+      qualityInt = DMWEB_QUALITY_UNKNOWN;
+    } else {
+      throw new IllegalArgumentException("Invalid quality string");
+    }
+
+    // Set quality integer
+    this.quality = qualityInt;
+  }
+
+  /**
+   * Returns the Ewon tag quality.
+   *
+   * @return the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getQuality() {
+    return quality;
+  }
+
+  /**
+   * Returns the vendor-specific quality information of the Ewon tag quality.
+   *
+   * @return the vendor-specific quality information of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getVendorSpecificQualityInformation() {
+    return (quality >> 8) & 0xFF;
+  }
+
+  /**
+   * Returns the major quality of the Ewon tag quality.
+   *
+   * @return the major quality of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getMajorQuality() {
+    return (quality >> 6) & 0x03;
+  }
+
+  /**
+   * Returns the sub-status of the Ewon tag quality.
+   *
+   * @return the sub-status of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getSubStatus() {
+    return (quality >> 2) & 0x0F;
+  }
+
+  /**
+   * Returns the limit status of the Ewon tag quality.
+   *
+   * @return the limit status of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getLimitStatus() {
+    return quality & 0x03;
+  }
+
+  /**
+   * Returns the Ewon tag quality as an Ignition {@link QualityCode} object.
+   *
+   * @return the Ewon tag quality as an Ignition {@link QualityCode} object
+   * @since 1.0.0
+   */
+  public QualityCode getQualityCode() {
+    QualityCode result;
+    switch (getMajorQuality()) {
+      case 0:
+        result = QualityCode.Bad;
+        break;
+      case 1:
+        result = QualityCode.Uncertain;
+        break;
+      case 3:
+        result = QualityCode.Good;
+        break;
+      default:
+        result = QualityCode.Bad_OutOfRange;
+        break;
+    }
+
+    // If quality is not good, check sub-status
+    if (result != QualityCode.Good) {
+      if (getSubStatus() == 1) {
+        result = QualityCode.Error_Configuration;
+      } else if (getSubStatus() == 3) {
+        result = QualityCode.Bad_Failure;
+      } else if (getSubStatus() == 6) {
+        result = QualityCode.Error_TimeoutExpired;
+      } else if (getSubStatus() == 7) {
+        result = QualityCode.Bad_Disabled;
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Returns the Ewon tag quality as an Ignition {@link DataQuality} object.
+   *
+   * @return the Ewon tag quality as an Ignition {@link DataQuality} object
+   * @see #getQualityCode()
+   * @since 1.0.0
+   */
+  public DataQuality getDataQuality() {
+    return DataQuality.getQualityFor(getQualityCode());
```

Comment:
method marked as deprecated. perhaps getQualityFor(int)
![image](https://github.com/hms-networks/IgnitionEwonConnector/assets/72526279/d66c3898-8a2a-4cb2-96e3-4aed71fa5180)

Reply:
This was used in previous connector versions, but I have updated it to directly convert to a corresponding value, similar to the above method.